src/NDItk/MultigroupTable/src/ctor.hpp

```diff
@@ -2,44 +2,56 @@
+  weights_(), total_(), xs_(), release_() {}
 
 /**
  *  @brief Constructor
  *
  *  @param[in] zaid           the zaid of the table
  *  @param[in] libname        the library name
- *  @param[in] source         the source date
  *  @param[in] process        the processing date
  *  @param[in] awr            the atomic weight ratio of the target
  *                            (with respect to the neutron mass)
- *  @param[in] weight         the atomic weight of the target
  *  @param[in] temperature    the temperature of the target
  *  @param[in] dilution       the dilution (aka sigma0)
  *  @param[in] structure      the primary group structure
  *  @param[in] outgoing       the outgoing particle group structures
+ *  @param[in] velocities     the group velocities
  *  @param[in] weights        the flux weights
+ *  @param[in] total          the total cross section
  *  @param[in] xs             the reaction cross section data
- *  @param[in] release        the average fission energy release data
+ *  @param[in] source         the source date (optional)
+ *  @param[in] weight         the atomic weight of the target (optional)
+ *  @param[in] release        the average fission energy release data (optional)
  */
-MultigroupTable( std::string zaid, std::string libname, std::string source,
-                 std::string process, double awr, double weight, double temperature,
+MultigroupTable( std::string zaid, std::string libname,
+                 std::string process, double awr, double temperature,
                  double dilution,
                  multigroup::EnergyGroupStructure structure,
                  std::vector< multigroup::EnergyGroupStructure > outgoing,
+                 multigroup::Velocities velocities,
                  multigroup::FluxWeights weigths,
                  multigroup::ReactionCrossSections xs,
+                 std::optional< std::string > source = std::nullopt,
+                 std::optional< double > weight = std::nullopt,
+                 std::optional< multigroup::TotalCrossSection > total = std::nullopt,
                  std::optional< multigroup::AverageFissionEnergyRelease > release = std::nullopt ) :
-    metadata_( std::move( zaid ), std::move( libname ), std::move( source ),
-               std::move( process ), awr, weight, temperature, dilution,
+    metadata_( std::move( zaid ), std::move( libname ),
+               std::move( process ), awr, temperature, dilution,
                structure.numberGroups(), generateOutgoingStructureMetadata( outgoing ),
-               xs.numberReactions() ),
+               xs.numberReactions(), std::move( source ), std::move( weight ),
+               std::nullopt, std::nullopt ),
     primary_structure_( std::move( structure ) ),
+    velocities_( std::move( velocities ) ),
     outgoing_structure_( std::move( outgoing ) ),
     weights_( std::move( weigths ) ),
     xs_( std::move( xs ) ),
     release_( std::move( release.value() ) ) {
```

Comment:
Should this have an if check since it's optional? Same with xs above.

Reply:
Yup.